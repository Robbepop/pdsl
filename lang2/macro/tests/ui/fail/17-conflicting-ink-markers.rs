#![feature(proc_macro_hygiene)]

use ink_lang2 as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    #[ink(event)] // We cannot have #[ink(event)] if we already have #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        fn new(&mut self) {}

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
