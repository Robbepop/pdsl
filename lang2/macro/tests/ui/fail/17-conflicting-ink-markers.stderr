error: invalid ink! attribute found for `#[ink(storage)]` struct
 --> $DIR/17-conflicting-ink-markers.rs:8:10
  |
8 |     #[ink(event)] // We cannot have #[ink(event)] if we already have #[ink(storage)]
  |          ^^^^^^^
